src/syncio.c

```diff
@@ -136,10 +136,10 @@ ssize_t syncReadLine(int fd, char *ptr, ssize_t size, long long timeout) {
             return nread;
         } else {
             *ptr++ = c;
-            *ptr = '\0';
             nread++;
         }
         size--;
     }
+    *ptr = '\0';
```

Comment:
Thanks. Although this is correct it changes the behavior on failure not to return a null terminated string. So I'm kind of reluctant to approve this.